[bug]: build error when building chan_sip

### Severity

Minor

### Versions

20.10.0

### Components/Modules

chan_sip

### Operating Environment

OS: Ubuntu 24.04.1 LTS
GCC: gcc (Ubuntu 13.2.0-23ubuntu4) 13.2.0

### Frequency of Occurrence

Constant

### Issue Description

Error on build

### Relevant log output

```shell
chan_sip.c:4286:24: error: conflicting types for ‘__sip_reliable_xmit’ due to enum/integer mismatch; have ‘enum sip_result(struct sip_pvt *, uint32_t,  int,  struct ast_str *, int,  int)’ {aka ‘enum sip_result(struct sip_pvt *, unsigned int,  int,  struct ast_str *, int,  int)’} [-Werror=enum-int-mismatch]
 4286 | static enum sip_result __sip_reliable_xmit(struct sip_pvt *p, uint32_t seqno, int resp, struct ast_str *data, int fatal, int sipmethod)
      |                        ^~~~~~~~~~~~~~~~~~~
chan_sip.c:1206:12: note: previous declaration of ‘__sip_reliable_xmit’ with type ‘int(struct sip_pvt *, uint32_t,  int,  struct ast_str *, int,  int)’ {aka ‘int(struct sip_pvt *, unsigned int,  int,  struct ast_str *, int,  int)’}
 1206 | static int __sip_reliable_xmit(struct sip_pvt *p, uint32_t seqno, int resp, struct ast_str *data, int fatal, int sipmethod);
```


### Asterisk Issue Guidelines

- [X] Yes, I have read the Asterisk Issue Guidelines

--- a/channels/chan_sip.c
+++ b/channels/chan_sip.c
@@ -1203,7 +1203,7 @@ static int get_address_family_filter(unsigned int transport);
 /*--- Transmitting responses and requests */
 static int sipsock_read(int *id, int fd, short events, void *ignore);
 static int __sip_xmit(struct sip_pvt *p, struct ast_str *data);
-static int __sip_reliable_xmit(struct sip_pvt *p, uint32_t seqno, int resp, struct ast_str *data, int fatal, int sipmethod);
+static enum sip_result __sip_reliable_xmit(struct sip_pvt *p, uint32_t seqno, int resp, struct ast_str *data, int fatal, int sipmethod);
 static void add_cc_call_info_to_response(struct sip_pvt *p, struct sip_request *resp);
 static int __transmit_response(struct sip_pvt *p, const char *msg, const struct sip_request *req, enum xmittype reliable);
 static int retrans_pkt(const void *data);

